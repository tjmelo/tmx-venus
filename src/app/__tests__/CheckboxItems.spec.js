import toCheckbox from '../modules/CheckboxItems';

const DOMCompare = "<input class='check' type='checkbox' value='test description' />";
document.body.innerHTML = DOMCompare;
const testElementToBeCompared = document.querySelector('input.check');

describe('Should get a param and render a checkbox', () => {
  it('Should render a checkbox', () => {
    const testToCheckbox = jest.fn();
    const testParamValue = 'test description';

    testToCheckbox.mockImplementation((argTest) => toCheckbox(argTest));

    // do
    testToCheckbox(testParamValue);

    // then
    expect(testToCheckbox).toHaveReturned();
    expect(testToCheckbox).toHaveBeenCalledWith(testParamValue);
    expect(testToCheckbox).toHaveBeenCalledTimes(1);
    expect(testToCheckbox).toHaveReturnedWith(testElementToBeCompared);

    // Expects fidelity with scope of function
    expect(testToCheckbox).toMatchSnapshot();
  });
});
