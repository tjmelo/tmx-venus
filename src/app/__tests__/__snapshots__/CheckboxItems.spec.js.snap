// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should get a param and render a checkbox Should render a checkbox 1`] = `
[MockFunction] {
  "calls": [
    [
      "test description",
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": <input
        class="check"
        type="checkbox"
        value="test description"
      />,
    },
  ],
}
`;
